contracts/icp/context-config/src/mutate.rs

```diff
@@ -282,3 +329,44 @@ fn revoke(
         Ok(())
     })
 }
+
+async fn update_proxy_contract(
+    signer_id: &ICSignerId,
+    context_id: ICContextId,
+) -> Result<(), String> {
+    let mut context = CONTEXT_CONFIGS.with(|configs| {
+        let configs = configs.borrow();
+        configs
+            .contexts
+            .get(&context_id)
+            .ok_or_else(|| "context does not exist".to_string())
+            .cloned()
+    })?;
+
+    // Get proxy canister ID
+    let proxy_canister_id = context
+        .proxy
+        .get(signer_id)
+        .map_err(|_| "unauthorized: Proxy capability required".to_string())?
+        .get_mut()
+        .clone();
+
+    // Get the proxy code
+    let proxy_code = CONTEXT_CONFIGS
+        .with(|configs| configs.borrow().proxy_code.clone())
+        .ok_or("proxy code not set")?;
+
+    // Update the proxy contract code
+    let install_args = InstallCodeArgument {
+        mode: ic_cdk::api::management_canister::main::CanisterInstallMode::Upgrade(None),
+        canister_id: proxy_canister_id,
+        wasm_module: proxy_code,
+        arg: candid::encode_one(&context_id).map_err(|e| format!("Encoding error: {}", e))?,
+    };
+
+    install_code(install_args)
+        .await
+        .map_err(|e| format!("Failed to update proxy contract: {:?}", e))?;
```

Comment:
but do we have permissions to do it from the context config? that seems like a massive security issue if it did

Reply:
yes, implemented the right way of checking the permission and preserving storage